nucleus/flashlight/framework/src/main/java/org/glassfish/flashlight/impl/client/ProbeProviderClassFileTransformer.java

```diff
@@ -323,7 +261,7 @@ public MethodVisitor visitMethod(int access, String name, String desc, String si
             MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
 
             FlashlightProbe probe = probes.get(makeKey(name, desc));
-            if (probe != null) {
+            if (probe != null && probe.isEnabled() && probe.getInvokerCount() > 0) {
```

Comment:
Important change to the way this class works: We decide if a method gets instrumented based on the state of the probe. As all state changes should trigger flag for update this is now safe to do.